tests/bluetooth/ctrl_sw_privacy_unit/src/main.c

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2020 Demant
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include <zephyr/types.h>
+#include <ztest.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#define CONFIG_BT_CTLR_RL_SIZE 8
+#define CONFIG_BT_CTLR_RPA_CACHE_SIZE 4
+#define CONFIG_BT_CTLR_PRIVACY 1
+#define CONFIG_BT_CTLR_SW_DEFERRED_PRIVACY 1
+#define CONFIG_BT_LOG_LEVEL 1
+
+#include "ll_sw/ull_filter.c"
+
+/*
+ * Unit test of SW deferred privacy data structure and related methods
+ * Tests the prpa_cache_add, prpa_cache_clear and prpa_cache_find functions
+ */
+
+#define BT_RAND_XX (rand() % 0xFF)
+
+#define BT_ADDR_INIT(P0, P1, P2, P3, P4, P5)                                   \
+	(&(bt_addr_t){ { P0, P1, P2, P3, P4, P5 } })
+
+#define BT_ADDR_LE_INIT(ADDRTYPE, P0, P1, P2, P3, P4, P5)                      \
+	(&(bt_addr_le_t){ ADDRTYPE, { { P0, P1, P2, P3, P4, P5 } } })
+
+#define IRK(P0, P1, P2, P3, P4, P5, P6, P7, P8, P9, P10, P11, P12, P13, P14,   \
+	    P15)                                                               \
+	({ P0, P1, P2, P3, P4, P5, P6, P7, P8, P9, P10, P11, P12, P13, P14,    \
+	   P15 })
+
+#define RAND_IRK                                                               \
+	(IRK(BT_RAND_XX, BT_RAND_XX, BT_RAND_XX, BT_RAND_XX, BT_RAND_XX,       \
```

Comment:
As discussed, this is dead code, please remove

Reply:
I will remove most of the macros